src/erlcloud_cur.erl

```diff
@@ -0,0 +1,133 @@
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer(), string()) ->
+    aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                cur_scheme        = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%% @doc
+%% Cost & Usage Report API:
+%% https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/describe-report-definitions.html
+%%
+%% `
+%% {ok, R} = erlcloud_cur:describe_report_definitions(Config).
+%% `
+%%
+-spec describe_report_definitions(aws_config()) -> {ok, map()} | {error, any()}.
+describe_report_definitions(Config) ->
+    describe_report_definitions(#{}, Config).
+
+
+%% `
+%% Opts = #{<<"MaxResults">> => 5},
+%% {ok, R} = erlcloud_cur:describe_report_definitions(Opts, Config).
+%% `
+
+-spec describe_report_definitions(map(), aws_config()) -> {ok, map()} | {error, any()}.
+describe_report_definitions(Options, Config) ->
+    request(Config, <<"DescribeReportDefinitions">>, Options).
+
+%%------------------------------------------------------------------------------
+%% Internal Functions
+%%------------------------------------------------------------------------------
+
+request(Config0, Operation, Input) ->
+    case erlcloud_aws:update_config(Config0) of
+        {ok, Config} ->
+            Body       = jsx:encode(Input),
+            Headers    = get_headers(Config, Operation, Body),
+            AwsRequest = #aws_request{service         = cur,
+                                      uri             = get_url(Config),
+                                      method          = post,
```

Comment:
Is it always POST? What about GETs?

Reply:
it is `post` for this service, even for `Describe*` functionality. (Probably the reason behind is that we actually have the query inside of the request body). If it's not the same is for other operations for `CUR` service (which is `DeleteReportDefinition` and `PutReportDefinition`(, I suggest to change it later when we add it

Reply:
it's a GET but via a POST with a body. so this is fine